src/Main.java

```diff
@@ -3,18 +3,46 @@
 public class Main {
 
     public static void main(String[] args) {
-        Scanner scanner = new Scanner(System.in);
-        initializeGame(scanner);
-        UI.start();
+        try (Scanner scanner = new Scanner(System.in)) {
+            int size = getBoardSize(scanner);
+            String difficulty = getDifficultyLevel(scanner);
+            Game.initialize(size, difficulty);
+            UI.start();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
     }
 
-    private static void initializeGame(Scanner s) {
-        System.out.println("Enter size of the board");
-        int size = Integer.parseInt(s.nextLine()); // add exception handling and input validation and while loop
 
-        System.out.println("Enter difficulty (easy, normal, hard)");
-        String difficulty = s.nextLine(); // add exception handling and input validation and while loop
+    private static int getBoardSize(Scanner scanner) {
+        int size = 0;
+        while (true) {
+            System.out.println("Enter size of the board (between 3 and 10):");
+            String input = scanner.nextLine();
+            try {
+                size = Integer.parseInt(input);
+                if (size >= 3 && size <= 10) {
```

Comment:
why only 10? let's do it 30

Reply:
just thought that a CLI game of 30 would be too overwhelming 😂😂
 will change it - on it!